java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -137,6 +139,10 @@ public boolean isNull(ProgramState ps, SymbolicValue val) {
     return NullConstraint.NULL.equals(ps.constraints.get(val));
   }
 
+  public boolean isConstrained(ProgramState ps, SymbolicValue val) {
+    return ps.constraints.containsKey(val);
+  }
+
   public Pair<ProgramState, ProgramState> assumeDual(ProgramState programState, Tree condition) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "assumeDual" is 26 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 12, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1142)